contracts/root/StakeManager.sol

```diff
@@ -1,33 +1,26 @@
 pragma solidity ^0.4.24;
 
-
 import { ERC20 } from "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
 import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
 import { Math } from "openzeppelin-solidity/contracts/math/Math.sol";
 
 
-import { AvlTree } from "../lib/AvlTree.sol";
 import { BytesLib } from "../lib/BytesLib.sol";
 import { ECVerify } from "../lib/ECVerify.sol";
 
 
 import { Lockable } from "../mixin/Lockable.sol";
 import { RootChainable } from "../mixin/RootChainable.sol";
 
-import { StakeManagerInterface } from "./StakeManagerInterface.sol";
-import { RootChain } from "./RootChain.sol";
+import { Validator } from "./Validator.sol";
+import { IStakeManager } from "./IStakeManager.sol";
 
 
-contract StakeManager is StakeManagerInterface, RootChainable, Lockable {
+contract StakeManager is IStakeManager, Validator, RootChainable, Lockable {
```

Comment:
Why extend `StakeManager` with `Validator` and not use a different contract? StakeManager will be big as there will be reward/slash logic will come too.

Reply:
In my view, `validatorId` is not slot position.

Reply:
IMO slashing, rewards should be done on a separate contract which communicates with root chain and stage manager both, also thinking about abstracting out the validator data(as our discussion in other comments).